include/mbedtls/fsio.h

```diff
@@ -0,0 +1,255 @@
+/**
+ * \file fsio.h
+ *
+ * \brief File read/write functions
+ *
+ *  Copyright (C) 2017, ARM Limited, All Rights Reserved
+ *  SPDX-License-Identifier: Apache-2.0
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License"); you may
+ *  not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ *  WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ *  This file is part of mbed TLS (https://tls.mbed.org)
+ */
+#ifndef MBEDTLS_FILE_H
+#define MBEDTLS_FILE_H
+
+#if !defined(MBEDTLS_CONFIG_FILE)
+#include "config.h"
+#else
+#include MBEDTLS_CONFIG_FILE
+#endif
+
+#include <stddef.h>
+#include <stdint.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#if defined(MBEDTLS_FS_IO) && !defined(MBEDTLS_FS_IO_ALT)
+/* If MBEDTLS_FS_IO is enabled then file IO functions should be made available
+ * via standard library or platform specific implementation or
+ * user defined alternative implementation. This file provides implementations
+ * based on serialization and standard library. */
+
+/**
+ * Structure returned by mbedtls_stat().
+ *
+ * \note Future versions of the library may add more fields at the end of this
+ *       structure.
+ */
+typedef struct mbedtls_stat_tag
+{
+    uint16_t    type;   /** File type */
+/** A file that can be accessed as a stream, i.e. opened with mbedtls_fopen().
+ */
+#define MBEDTLS_FSIO_DT_FILE    0
+/** A directory that can be opened with mbedtls_opendir(). */
+#define MBEDTLS_FSIO_DT_DIR     1
+/** A directory entry that is recognized neither as a stream file nor as a
+ * directory. */
+#define MBEDTLS_FSIO_DT_OTHER   2
+} mbedtls_stat_t;
+
+/* Default and alternative implementation specific interfaces. */
+
+/* MBEDTLS_SERIALIZE_C replaces Standard library FS implementation */
+#if defined(MBEDTLS_SERIALIZE_C)
+
+/**
+ * file handle.
+ *
+ */
+typedef int32_t mbedtls_file_t;
+#define MBEDTLS_FILE_INVALID    -1
+
+/**
+ * dir handle.
+ *
+ */
+typedef int32_t mbedtls_dir_t;
+#define MBEDTLS_DIR_INVALID     -1
+
+/**
+ * Definition of reference for seek offset used by mbedtls_fseek().
+ */
+#define MBEDTLS_SEEK_SET    0
+#define MBEDTLS_SEEK_CUR    1
+#define MBEDTLS_SEEK_END    2
+
+/**
+ * \brief          Read file.
+ *
+ * \param ptr      Pointer to output buffer
```

Comment:
Style: By now, we are trying to follow the documentation convention of always writing full sentences, I think.